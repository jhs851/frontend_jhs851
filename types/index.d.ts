export interface User {
  id: number
  name: string
  avatar: string
}

export interface Room {
  id: number
  user_id: number
  local_name: string
  unread_message_count: number
  last_message_id?: number
}

export interface RelatedRoom extends Room {
  user: User
  last_message?: Message
}

export interface Message {
  id: number
  user_id: number
  room_id: number
  message?: string
  asset?: string
  created_at: string
}

export interface RelatedMessage extends Message {
  user: User
  room: Room
}

export interface VirtualMessage extends Omit<Message, 'id'> {
  virtual_id: string
}

export interface Asset {
  id: number
  path: string
}
