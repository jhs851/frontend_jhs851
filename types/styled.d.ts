import 'styled-components'
import { Colors } from 'types/theme'

declare module 'styled-components' {
  export interface DefaultTheme extends Colors {}
}
