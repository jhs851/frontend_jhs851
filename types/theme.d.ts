export interface CommonColors {
  readonly black: string
  readonly white: string
  readonly grey1: string
  readonly coolGrey: string
  readonly paleGrey: string
  readonly battleshipGrey: string
  readonly charcoalGrey: string
  readonly charcoalGrey2: string
  readonly paleLilac: string
  readonly foreground: string
  readonly background: string
  readonly black10: string
}

export interface AccentColors {
  readonly primary: string
  readonly secondary: string
  readonly error: string
}

export interface Colors extends CommonColors, AccentColors {}

export type Theme = {
  light: Colors
  dark: Colors
}

export interface ThemeContextState {
  dark: boolean
  toggleTheme: () => void
  colors: Colors
}
