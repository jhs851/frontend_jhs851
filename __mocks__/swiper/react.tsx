import React from 'react'

const mock: React.FC = ({ children }) => <div>{children}</div>

export default mock
