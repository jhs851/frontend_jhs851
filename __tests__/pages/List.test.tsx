import { render } from '@testing-library/react'
import List from 'pages/list'
import {
  rooms as mockRooms,
  users as mockUsers,
  messages as mockMessages,
} from 'utils/mocks'
import { RelatedRoom as RRoom } from 'types'

describe('채팅 리스트 페이지', () => {
  it('렌더링이 올바르게 된다', async () => {
    const rooms: RRoom[] = mockRooms.map(room => ({
      ...room,
      user: mockUsers.find(user => user.id === room.user_id) || mockUsers[0],
      last_message: mockMessages.find(
        message => message.id === room.last_message_id,
      ),
    }))

    expect(render(<List rooms={rooms} />).container).toMatchSnapshot()
  })
})
