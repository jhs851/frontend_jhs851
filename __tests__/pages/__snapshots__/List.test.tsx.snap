// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`채팅 리스트 페이지 렌더링이 올바르게 된다 1`] = `
.c0 {
  padding: 10px 0;
}

.c6 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 16px;
  font-weight: 700;
  -webkit-letter-spacing: -0.2px;
  -moz-letter-spacing: -0.2px;
  -ms-letter-spacing: -0.2px;
  letter-spacing: -0.2px;
}

.c7 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 13px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  overflow: hidden;
  text-overflow: ellipsis '..';
  white-space: nowrap;
}

.c10 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 11px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

.c13 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 10px;
  font-weight: 700;
  -webkit-letter-spacing: -0.08px;
  -moz-letter-spacing: -0.08px;
  -ms-letter-spacing: -0.08px;
  letter-spacing: -0.08px;
}

.c1 {
  height: 74px;
  padding: 0 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-flex: 0 0 56px;
  -ms-flex: 0 0 56px;
  flex: 0 0 56px;
  border-radius: 50%;
  overflow: hidden;
  margin-right: 15px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.c8 {
  margin-top: 3px;
}

.c9 {
  height: 100%;
  margin-left: 13px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c11 {
  opacity: 0.4;
}

.c12 {
  width: 18px;
  height: 18px;
  margin-top: 6px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <section
    class="c0"
  >
    <a
      class="c1"
      data-testid="room-component"
      href="/room/1"
    >
      <div
        class="c2"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c3"
        />
      </div>
      <div
        class="c4"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            letter-spacing="-0.2"
            size="16"
          >
            장만월 사장님
          </p>
          <p
            class="c7 c8"
            color="coolGrey"
            size="13"
          >
            어딘데 출근 안하니, 죽고싶니?
          </p>
        </div>
        <div
          class="c9"
        >
          <p
            class="c10 c11"
            letter-spacing="0"
            size="11"
          >
            09:32
          </p>
          <div
            class="c12"
          >
            <p
              class="c13"
              color="white"
              letter-spacing="-0.08"
              size="10"
            >
              2
            </p>
          </div>
        </div>
      </div>
    </a>
    <a
      class="c1"
      data-testid="room-component"
      href="/room/2"
    >
      <div
        class="c2"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c3"
        />
      </div>
      <div
        class="c4"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            letter-spacing="-0.2"
            size="16"
          >
            신정근 바텐더
          </p>
          <p
            class="c7 c8"
            color="coolGrey"
            size="13"
          >
            오시는 길에 와인 몇병만 사다주세요.
          </p>
        </div>
        <div
          class="c9"
        >
          <p
            class="c10 c11"
            letter-spacing="0"
            size="11"
          >
            02:34
          </p>
        </div>
      </div>
    </a>
    <a
      class="c1"
      data-testid="room-component"
      href="/room/3"
    >
      <div
        class="c2"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c3"
        />
      </div>
      <div
        class="c4"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            letter-spacing="-0.2"
            size="16"
          >
            이미라 의사
          </p>
          <p
            class="c7 c8"
            color="coolGrey"
            size="13"
          >
            휴가 잘 보내고 계신가요? 다름이 아니라 지금 검사 결과가 나와서 연락드립니다.
          </p>
        </div>
        <div
          class="c9"
        >
          <p
            class="c10 c11"
            letter-spacing="0"
            size="11"
          >
            Saturday
          </p>
        </div>
      </div>
    </a>
    <a
      class="c1"
      data-testid="room-component"
      href="/room/4"
    >
      <div
        class="c2"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c3"
        />
      </div>
      <div
        class="c4"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            letter-spacing="-0.2"
            size="16"
          >
            구찬성 지배인
          </p>
          <p
            class="c7 c8"
            color="coolGrey"
            size="13"
          >
            아 휴가셨군요. 약속은 다음으로 미루시죠!
          </p>
        </div>
        <div
          class="c9"
        >
          <p
            class="c10 c11"
            letter-spacing="0"
            size="11"
          >
            Friday
          </p>
        </div>
      </div>
    </a>
    <a
      class="c1"
      data-testid="room-component"
      href="/room/5"
    >
      <div
        class="c2"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c3"
        />
      </div>
      <div
        class="c4"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            letter-spacing="-0.2"
            size="16"
          >
            노준석 총지배인
          </p>
          <p
            class="c7 c8"
            color="coolGrey"
            size="13"
          >
            휴가에서 언제 돌아오시는지요. 돌아오시면 회포 한번 푸시죠.
          </p>
        </div>
        <div
          class="c9"
        >
          <p
            class="c10 c11"
            letter-spacing="0"
            size="11"
          >
            Thursday
          </p>
        </div>
      </div>
    </a>
    <a
      class="c1"
      data-testid="room-component"
      href="/room/6"
    >
      <div
        class="c2"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c3"
        />
      </div>
      <div
        class="c4"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            letter-spacing="-0.2"
            size="16"
          >
            김유나 인턴
          </p>
          <p
            class="c7 c8"
            color="coolGrey"
            size="13"
          >
            304호 키를 잃어버렸어요 어떻게 해야하죠 ㅠ
          </p>
        </div>
        <div
          class="c9"
        >
          <p
            class="c10 c11"
            letter-spacing="0"
            size="11"
          >
            Wednesday
          </p>
        </div>
      </div>
    </a>
    <a
      class="c1"
      data-testid="room-component"
      href="/room/7"
    >
      <div
        class="c2"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c3"
        />
      </div>
      <div
        class="c4"
        style="opacity: 0; transform: translateX(-30px) translateZ(0);"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            letter-spacing="-0.2"
            size="16"
          >
            구현모
          </p>
          <p
            class="c7 c8"
            color="coolGrey"
            size="13"
          >
            술먹자
          </p>
        </div>
        <div
          class="c9"
        >
          <p
            class="c10 c11"
            letter-spacing="0"
            size="11"
          >
            Tuesday
          </p>
        </div>
      </div>
    </a>
  </section>
</div>
`;
